packages/daemon/src/app/controller/job.ts

```diff
@@ -17,6 +19,40 @@ export class JobController extends BaseEventController {
   @inject('pluginManager')
   pluginManager: PluginManager;
 
+  private async setupTracer(job: JobEntity) {
+    const logPath = join(constants.PIPCOOK_RUN, job.id, 'logs');
+    const stdoutFile = join(logPath, 'stdout.log');
+    const stderrFile = join(logPath, 'stderr.log');
+    await ensureDir(logPath);
+    await [
+      ensureFile(stdoutFile),
+      ensureFile(stderrFile)
+    ];
+    const tracer = await this.traceManager.create({ stdoutFile, stderrFile });
+    return tracer;
+  }
+
+  private makeParam(pipeline: PipelineEntity, jobParams?: IJobParam[]) {
```

Comment:
Shall we move this to service?

Reply:
This function should only be called within job process procedure.